TRACING-4456: improve documentation about zPages extension (#82621)
openshift/openshift-docs: observability/otel/otel-collector/otel-collector-extensions.adoc

        protocols:
          http: {}
    exporters:
      debug:

    service:
      extensions: [zpages]
      pipelines:
        traces:
          receivers: [otlp]
          exporters: [debug]
# ...
----

<1> Specify the HTTP endpoint for serving the zPages extension. The default is `localhost:55679`.

> The {OTELOperator} does not expose this route. To access the HTTP endpoint, forward the port:

Reviewer: This to me reads like it should be in an admonition block. 

[NOTE]
====
TEXT
====

[source,terminal]
----
$ oc port-forward pod/$(oc get pod -l app.kubernetes.io/name=instance-collector -o=jsonpath='{.items[0].metadata.name}') 55679
----

> To get the zPages information, you can visit `http://localhost:55679/debug/servicez` in your browser.

Author: The link will not work for you because you need to execute all the commands described previously.
It is not an additional resource; it becomes available when the user runs the previous commands. This URL needs to be documented clearly.

> To get the zPages information, you can visit `http://localhost:55679/debug/servicez` in your browser.

Reviewer: Thanks for clarifying, @iblancasa . This is where the REFERENCE module gets tricky as procedural steps are being discussed. As annoying as this is, maybe having separate procedure and reference module would make for a more comprehensive understanding of what steps are needed to reach an end goal?

> To get the zPages information, you can visit `http://localhost:55679/debug/servicez` in your browser.

Reviewer 2: I think I can offer a suggestion to rewrite this in a more referential tone.

> To get the zPages information, you can visit `http://localhost:55679/debug/servicez` in your browser.

Reviewer: See https://redhat-documentation.github.io/modular-docs/#reference-module-guidelines . A CONCEPT AND REFERENCE module could also work, but instead of procedure steps, you could write a scenario based on an example.

> To get the zPages information, you can visit `http://localhost:55679/debug/servicez` in your browser.

Reviewer 2: @iblancasa and @dfitzmau, here is my suggestion:
https://github.com/openshift/openshift-docs/pull/82621/files#r1782949553

> Now, you can visit `http://localhost:55679/debug/servicez` in yor browser to get the zPages information.

Reviewer 2: To avoid this appearing like a procedure, you could rephrase this sentence as follows:
```suggestion
To get the zPages information, you can visit `http://localhost:55679/debug/servicez` in your browser.
```

The Collector offers the following zPages for diagnostics:

* ServiceZ: provides an overview of the collector services and offers links to the `pipelinez`, `extensionz`, and `featurez` zPages. This page also displays information about the build version and runtime. Example URL: `http://localhost:55679/debug/servicez`.
* PipelineZ: offers detailed information about the active pipelines within the collector. It shows the pipeline type, whether data is modified, and the associated receivers, processors, and exporters for each pipeline. Example URL: `http://localhost:55679/debug/pipelinez`.
* ExtensionZ: shows the currently active extensions within the collector. Example URL: `http://localhost:55679/debug/extensionz`.
* FeatureZ: displays the feature gates enabled in the collector along with their status and description. Example URL: `http://localhost:55679/debug/featurez`.

> * TraceZ: allows users to view spans categorized by latency, such as in time ranges like `(0us, 10us, 100us, 1ms, 10ms, 100ms, 1s, 10s, 1m]`. This page also allows for quick inspection of error samples. Example URL: `http://localhost:55679/debug/tracez`.

Author: > BTW, does the ServiceZ page not contain a link to the TraceZ page?

No, it doesn't contain that (I had to double check).

![Screenshot from 2024-09-30 18-14-12](https://github.com/user-attachments/assets/45dbd0dd-f31d-4269-a7f1-80328cd27352)

>         protocols:

Reviewer: A sentence explaining the YAML would be useful as it seems a bit dropped in to the module.

Author: It's just an example configuration to have something users can copy, paste, and test.

Reviewer: NP. The heading might be enough to explain the purpose of the YAML. Feel free to send this for merging.

Reviewer: @max-cx , are you OK with this to move forward. I'm happy with the updates.

          http: {}
    exporters:
      debug:

    service:
      extensions: [zpages]
      pipelines:
        traces:
          receivers: [otlp]
          exporters: [debug]
# ...
----

> <1> Specify the HTTP endpoint for serving the zPages extension. The default is `localhost:55679`.

Reviewer: ```suggestion
<1> Specifies the HTTP endpoint for serving the zPages extension. The default is `localhost:55679`.
```

This would take the procedural voice out of this reference module.

[NOTE]
====
The {OTELOperator} does not expose this route. To access the HTTP endpoint, forward the port:

[source,terminal]
----
$ oc port-forward pod/$(oc get pod -l app.kubernetes.io/name=instance-collector -o=jsonpath='{.items[0].metadata.name}') 55679
----

To get the zPages information, you can visit `http://localhost:55679/debug/servicez` in your browser.

> ====

Author: I would prefer to keep the URL even when it is mentioned after this. I think it makes it easier to follow for users but I don't have a strong opinion.

> ====

Reviewer 2: @iblancasa, my suggestion for dropping this sentence with the URL here has two reasons:

1.
If you drop this sentence with the URL, you get a clear structure for this section:
1. Intro paragraph.
2. CR YAML block.
3. Port-forwarding admonition.
4. List of zPages with URLs.

2.
You don't have to worry about creating an additional two-step procedure module file that has two steps: 1) Run the port-forwarding command, and 2) Visit the URL in your browser.

> The Collector offers the following zPages for diagnostics:

Reviewer 2: ```suggestion
The Collector provides the following zPages for diagnostics:
```
